packages/openid4vc-client/src/OpenId4VcClientService.ts

```diff
@@ -134,6 +160,42 @@ export class OpenId4VcClientService {
     }
   }
 
+  public generateCodeVerifier(): string {
+    return randomStringForEntropy(256)
+  }
+
+  public async generateAuthorizationUrl(options: GenerateAuthorizationUrlOptions) {
+    this.logger.debug('Generating authorization url')
+
+    if (!options.scope || options.scope.length === 0) {
+      throw new AriesFrameworkError(
+        'Only scoped based authorization requests are supported at this time. Please provide at least one scope'
+      )
+    }
+
+    const client = await OpenID4VCIClient.initiateFromURI({
+      issuanceInitiationURI: options.initiationUri,
+      flowType: AuthzFlowType.AUTHORIZATION_CODE_FLOW,
+    })
+
+    const hashed = sha256(TypedArrayEncoder.fromString(options.codeVerifier))
+    const base64Url = TypedArrayEncoder.toBase64URL(Buffer.from(hashed))
+
+    this.logger.debug('Converted code_verifier to code_challenge', {
+      codeVerifier: options.codeVerifier.toString(),
+      sha256: hashed.toString(),
+      base64Url: base64Url,
+    })
+
+    return client.createAuthorizationRequestUrl({
+      clientId: options.clientId,
+      codeChallengeMethod: CodeChallengeMethod.SHA256,
```

Comment:
Could the authorization server support other algorithms than sha256? Or is this generally the one to support?

Reply:
In theory the code challenge method can also be set to TEXT where a plain text string is used instead. This basically defeats the purpose and according to my understanding it's not recommended or common. So far I've not come across auth servers that support it.